### PR TITLE
Fix settings CRS for new GRASS mapsets

### PR DESCRIPTION
**Pull request by nyalldawson**
Don't force the selected CRS to a (lossy) proj string, just use the CRS's WKT directly

This avoids losing projection information from some selected projections when creating a new GRASS mapset (eg EPSG:3111)

### CODE REVIEW

**Comment by nyalldawson** [comment]
View PR with whitespace hidden for ease of review...

File: src/plugins/grass/qgsgrassnewmapset.cpp [hunk 1]
```diff
@@ -399,64 +399,29 @@ void QgsGrassNewMapset::setGrassProjection()
...
-
-    if ( errcode != OGRERR_NONE )
-    {
-      QgsDebugMsgLevel( QString( "OGR can't parse PROJ-style parameter string:\n%1\nOGR Error code was %2" ).arg( proj4 ).arg( errcode ), 2 );
-
-      mCellHead.proj = PROJECTION_XY;
-      mCellHead.zone = 0;
-      mProjInfo = nullptr;
-      mProjUnits = nullptr;
+      // There was a bug in GRASS, it is present in 6.0.x line
```
Comment (lbartoletti) [inline src/plugins/grass/qgsgrassnewmapset.cpp:412]: Hasn't been fixed. We use only GRASS >= 7 now, isn't it?

File: src/plugins/grass/qgsgrassnewmapset.cpp [hunk 2]
```diff
@@ -399,64 +399,29 @@ void QgsGrassNewMapset::setGrassProjection()
...
+      int ret = GPJ_wkt_to_grass( &mCellHead, &mProjInfo, &mProjUnits, wkt.toUtf8().constData(), 0 );
+      // Note: It seems that GPJ_osr_to_grass()returns always 1,
+      //   -> test if mProjInfo was set
+
+      Q_UNUSED( ret )
+      QgsDebugMsgLevel( QString( "ret = %1" ).arg( ret ), 2 );
+      QgsDebugMsgLevel( QString( "mProjInfo = %1" ).arg( QString::number( ( qulonglong )mProjInfo, 16 ).toLocal8Bit().constData() ), 2 );
```
Comment (lbartoletti) [inline src/plugins/grass/qgsgrassnewmapset.cpp:419]:
```suggestion
      QgsDebugMsgLevel( QString( "mProjInfo = %1" ).arg( QString::number( static_cast<qulonglong>( mProjInfo ), 16 ).toLocal8Bit().constData() ), 2 );
```

File: src/plugins/grass/qgsgrassnewmapset.cpp [hunk 3]
```diff
@@ -399,64 +399,29 @@ void QgsGrassNewMapset::setGrassProjection()
...
-    {
-      QgsDebugMsgLevel( QString( "OGR can't parse PROJ-style parameter string:\n%1\nOGR Error code was %2" ).arg( proj4 ).arg( errcode ), 2 );
-
-      mCellHead.proj = PROJECTION_XY;
-      mCellHead.zone = 0;
-      mProjInfo = nullptr;
-      mProjUnits = nullptr;
+      // There was a bug in GRASS, it is present in 6.0.x line
+      int ret = GPJ_wkt_to_grass( &mCellHead, &mProjInfo, &mProjUnits, wkt.toUtf8().constData(), 0 );
+      // Note: It seems that GPJ_osr_to_grass()returns always 1,
```
Comment (lbartoletti) [inline src/plugins/grass/qgsgrassnewmapset.cpp:414]:
```suggestion
      // Note: It seems that GPJ_osr_to_grass() returns always 1,
```

Still true?

File: src/plugins/grass/qgsgrassnewmapset.cpp [hunk 4]
```diff
@@ -399,64 +399,29 @@ void QgsGrassNewMapset::setGrassProjection()
...
+      //   -> test if mProjInfo was set
```
Comment (lbartoletti) [inline src/plugins/grass/qgsgrassnewmapset.cpp:415]: Have you tested it?

File: src/plugins/grass/qgsgrassnewmapset.cpp [hunk 5]
```diff
@@ -399,64 +399,29 @@ void QgsGrassNewMapset::setGrassProjection()
...
   {
-    QgsDebugMsgLevel( QString( "proj4 = %1" ).arg( proj4.toLocal8Bit().constData() ), 3 );
+    const QString wkt = crs.toWkt( Qgis::CrsWktVariant::Preferred );
+    QgsDebugMsgLevel( QStringLiteral( "wkt = %1" ).arg( crs.toWkt( Qgis::CrsWktVariant::Preferred ) ), 3 );
 
-    OGRSpatialReferenceH hCRS = nullptr;
-    hCRS = OSRNewSpatialReference( nullptr );
-    int errcode;
+    // Note: GPJ_osr_to_grass() defaults in PROJECTION_XY if projection
+    //       cannot be set
```
Comment (lbartoletti) [inline src/plugins/grass/qgsgrassnewmapset.cpp:408]:
```suggestion
```

Doesn't seem used in your new code

File: src/plugins/grass/qgsgrassnewmapset.cpp (hunk 3 above)
Comment (nyalldawson) [reply src/plugins/grass/qgsgrassnewmapset.cpp:414]: @lbartoletti that's all existing comments and code -- I don't feel confident removing those notes.

File: src/plugins/grass/qgsgrassnewmapset.cpp (hunk 1 above)
Comment (nyalldawson) [reply src/plugins/grass/qgsgrassnewmapset.cpp:412]: I'm not sure -- it's an existing note which I'm not comfortable removing

**Review by alexbruy — APPROVED** [review]

**Pull request merged**